R/helpers.R

```diff
@@ -0,0 +1,419 @@
+#' Helper function which calculates p-value via chi-square or fisher
+#'
+#' @param df <`tbl_df`> Dataframe that has variable and treatment columns of interest
+#' @param var <`character(1)`> Name of variable column
+#' @param treatment <`character(1)`> Name of treatment column
+#'
+#' @examples
+#'
+#' p_chi_fisher(mtcars, var = "am", treatment = "vs")
+#'
+#' @return <`numeric(1)`> p-value
+#'
+#' @noRd
+p_chi_fisher <- function(df, var, treatment) {
+
+  chisq_wrapper <- function(var, df, treatment) {
+    table <- stats::chisq.test(as.factor(df[[var]]), as.factor(df[[treatment]]))
+    return(table$p.value)
+  }
+
+  fisher_wrapper <- function(var, df, treatment) {
+    table <- stats::fisher.test(as.factor(df[[var]]), as.factor(df[[treatment]]), simulate.p.value = TRUE)
+    return(table$p.value)
+  }
+
+  chisq_wrapper <- purrr::quietly(chisq_wrapper)
+  chisq <- chisq_wrapper(var, df, treatment)
+
+  if (length(chisq$warnings) == 0) {
+    return(chisq$result)
+  } else {
+    return(fisher_wrapper(var, df, treatment))
+  }
+
+}
+
+#' Helper function which calculates p-value via anova
+#'
+#' @inheritParams p_chi_fisher
+#'
+#' @return <`numeric(1)`> p-value
+#'
+#' @import dplyr
+#' @import stats
+#'
+#' @noRd
+p_anova <- function(df, var, treatment) {
+
+  form <- stats::as.formula(paste0(var, " ~ ", treatment))
+  lm(form, df) %>%
+    anova() %>%
+    pull(`Pr(>F)`) %>%
+    purrr::pluck(1)
+
+}
+
+#' Helper function which calculates p-value via Kruskal-Wallis Rank Sum Test
+#'
+#' @inheritParams p_chi_fisher
+#'
+#' @return <`numeric(1)`> p-value
+#'
+#' @noRd
+p_kruskal <- function(df, var, treatment) {
+
+  stats::kruskal.test(df[[var]], df[[treatment]]) %>%
+    purrr::pluck("p.value")
+
+}
+
+#' Helper function which calculates p-value via chi-square with weights
+#'
+#' @param df <`tbl_df`> Dataframe that has variable and treatment columns of interest
+#' @param var <`character(1)`> Name of variable column
+#' @param treatment <`character(1)`> Name of treatment column
+#' @param weight_var <`character(1)`> Name of weight column
+#'
+#' @return <`numeric(1)`> p-value
+#'
+#' @noRd
+p_chi_weighted <- function(df, var, treatment, weight_var) {
+
+  weight_var <- sym(weight_var)
```

Comment:
is there a reason why the `quo_name` and `sym` functions are used here and not in the other functions?

Reply:
I don't know why I'm quoting `weight_var` here. In the weighted function `var` and `treatment` must have needed to be quoted, so they were being passed into this function as quoted. I'm just going to remove this function for now as it's inconsistent that this function is in here, but weighted anova is not.